File: lib/gamechanger/commands/refresh.rb
# frozen_string_literal: true

module Gamechanger
  module Commands
    # `gamechanger refresh` — force re-sync of latest game data.
    #
    # Output respects --format: 'human' (default) prints a green count line,
    # 'json' prints a parseable JSON object for piping into other tools.
    class Refresh < Base
      def call
        run_command do
          config = load_config!
          with_storage(season: config.season) do |storage|
            shell.say 'Syncing games from Gamechanger...', :cyan
            result = Syncer.new(config, storage).run(force: true)
            if options[:format] == 'json'
              require 'json' # opt-in output format; keep json off the startup path
              puts JSON.generate({ games: result.games, outings: result.outings, at_bats: result.at_bats })
            else
              shell.say format_human(result), :green
            end
          end
        end
      end

      private

      def format_human(result)
        games   = result.games
        outings = result.outings
        at_bats = result.at_bats
        "#{games} game#{'s' unless games == 1}, " \
          "#{outings} outing#{'s' unless outings == 1}, " \
          "#{at_bats} at-bat#{'s' unless at_bats == 1} updated."
      end
    end
  end
end

File: internal/commands/refresh.go
package commands

import (
	"encoding/json"
	"fmt"

	"github.com/spf13/cobra"

	"github.com/joshrpowell/gamechanger-cli/internal/client"
	"github.com/joshrpowell/gamechanger-cli/internal/gcerr"
	"github.com/joshrpowell/gamechanger-cli/internal/store"
	syncpkg "github.com/joshrpowell/gamechanger-cli/internal/sync"
)

func newRefreshCmd(opts *rootOpts) *cobra.Command {
	var refreshFormat string
	cmd := &cobra.Command{
		Use:   "refresh",
		Short: "Sync latest game data from Gamechanger",
		Args:  cobra.NoArgs,
		RunE: func(cmd *cobra.Command, _ []string) error {
			cfg, err := opts.loadConfig()
			if err != nil {
				return err
			}
			if !cfg.Configured() {
				return gcerr.Configf("not configured. Run `gamechanger setup` first")
			}

			dir, err := opts.configDirOrDefault()
			if err != nil {
				return err
			}
			st, err := store.OpenAt(cmd.Context(), dir, cfg.Season)
			if err != nil {
				return err
			}
			defer st.Close()

			cli := client.New(cfg)
			syncer := &syncpkg.Syncer{
				Config: cfg,
				Client: cli,
				Store:  st,
			}

			fmt.Fprintln(cmd.ErrOrStderr(), "Syncing games from Gamechanger...")
			result, err := syncer.Run(cmd.Context(), true)
			if err != nil {
				return err
			}

			return printRefreshResult(cmd, refreshFormat, result)
		},
	}
	cmd.Flags().StringVar(&refreshFormat, "format", "human",
		"Output format: human | json")
	return cmd
}

func printRefreshResult(cmd *cobra.Command, format string, result store.SyncResult) error {
	out := cmd.OutOrStdout()
	if format == "json" {
		return json.NewEncoder(out).Encode(result)
	}
	fmt.Fprintf(out, "%s, %s, %s updated.\n",
		pluralize(result.Games, "game"),
		pluralize(result.Outings, "outing"),
		pluralize(result.AtBats, "at-bat"),
	)
	return nil
}

func pluralize(n int, noun string) string {
	if n == 1 {
		return fmt.Sprintf("%d %s", n, noun)
	}
	return fmt.Sprintf("%d %ss", n, noun)
}
